src/basic/REDUCE3_INT-OMPTarget.cpp

```diff
@@ -62,70 +62,35 @@ void REDUCE3_INT::runOpenMPTargetVariant(VariantID vid, size_t tune_idx)
 
   } else if ( vid == RAJA_OpenMPTarget ) {
 
-    if (tune_idx == 0) {
-
-      startTimer();
-      for (RepIndex_type irep = 0; irep < run_reps; ++irep) {
-
-        RAJA::ReduceSum<RAJA::omp_target_reduce, Int_type> vsum(m_vsum_init);
-        RAJA::ReduceMin<RAJA::omp_target_reduce, Int_type> vmin(m_vmin_init);
-        RAJA::ReduceMax<RAJA::omp_target_reduce, Int_type> vmax(m_vmax_init);
-
-        RAJA::forall<RAJA::omp_target_parallel_for_exec<threads_per_team>>(
-          RAJA::RangeSegment(ibegin, iend), [=](Index_type i) {
-          REDUCE3_INT_BODY_RAJA;
-        });
-
-        m_vsum += static_cast<Int_type>(vsum.get());
-        m_vmin = RAJA_MIN(m_vmin, static_cast<Int_type>(vmin.get()));
-        m_vmax = RAJA_MAX(m_vmax, static_cast<Int_type>(vmax.get()));
-
-      }
-      stopTimer();
-
-    } else if (tune_idx == 1) {
-
-      startTimer();
-      for (RepIndex_type irep = 0; irep < run_reps; ++irep) {
-
-        Int_type tvsum = m_vsum_init;
-        Int_type tvmin = m_vmin_init;
-        Int_type tvmax = m_vmax_init;
+    startTimer();
+    for (RepIndex_type irep = 0; irep < run_reps; ++irep) {
 
-        RAJA::forall<RAJA::omp_target_parallel_for_exec<threads_per_team>>(
-          RAJA::RangeSegment(ibegin, iend),
-          RAJA::expt::Reduce<RAJA::operators::plus>(&tvsum),
-          RAJA::expt::Reduce<RAJA::operators::minimum>(&tvmin),
-          RAJA::expt::Reduce<RAJA::operators::maximum>(&tvmax),
-          [=](Index_type i, Int_type& vsum, Int_type& vmin, Int_type& vmax) {
-            REDUCE3_INT_BODY;
-          }
-        );
+      Int_type tvsum = m_vsum_init;
+      Int_type tvmin = m_vmin_init;
+      Int_type tvmax = m_vmax_init;
 
-        m_vsum += static_cast<Int_type>(tvsum);
-        m_vmin = RAJA_MIN(m_vmin, static_cast<Int_type>(tvmin));
-        m_vmax = RAJA_MAX(m_vmax, static_cast<Int_type>(tvmax));
+      RAJA::forall<RAJA::omp_target_parallel_for_exec<threads_per_team>>(
+        RAJA::RangeSegment(ibegin, iend),
+        RAJA::expt::Reduce<RAJA::operators::plus>(&tvsum),
+        RAJA::expt::Reduce<RAJA::operators::minimum>(&tvmin),
+        RAJA::expt::Reduce<RAJA::operators::maximum>(&tvmax),
+        [=](Index_type i, Int_type& vsum, Int_type& vmin, Int_type& vmax) {
+          REDUCE3_INT_BODY;
+        }
+      );
 
-      }
-      stopTimer();
+      m_vsum += static_cast<Int_type>(tvsum);
+      m_vmin = RAJA_MIN(m_vmin, static_cast<Int_type>(tvmin));
+      m_vmax = RAJA_MAX(m_vmax, static_cast<Int_type>(tvmax));
```

Comment:
I'm sure there's a good reason, but why are we doing min and max operations outside of the forall here?

Reply:
Technically, it shouldn't matter when the kernel runs multiple times, but we want to make sure we are not just seeing the result from the last run through the kernel. This change was not made in this PR. It's been like that since the beginning.

Reply:
Ok, this looks fine because we're using the macros on integer types. If this were inside a lambda, we'd need to use `.min()` on the `ValOp` type in the near future.

Reply:
Yup. Those changes will be coming.